src/quo/components/inputs/address_input/view.cljs

```diff
@@ -55,124 +54,135 @@
+                                      (when (and (not address?)
+                                                 (not ens?)
+                                                 on-detect-unclassified)
+                                        (on-detect-unclassified text)))))
+        set-value                (rn/use-callback
+                                  (fn [new-value]
+                                    (reset! value new-value)
+                                    (on-change new-value)
+                                    (trigger-render-value new-value)))
+        on-paste                 (rn/use-callback
+                                  (fn []
+                                    (clipboard/get-string
+                                     (fn [clipboard]
+                                       (when-not (empty? clipboard)
+                                         (set-value clipboard))))))
+        on-clear                 (rn/use-callback
+                                  (fn []
+                                    (set-value "")
+                                    (set-status (if focused? :active :default))
+                                    (when on-change-text
+                                      (on-change-text ""))
+                                    (when on-clear
+                                      (on-clear)))
+                                  [focused?])
+        on-clear                 (rn/use-callback
+                                  (fn []
+                                    (set-value "")
+                                    (set-status (if focused? :active :default))
+                                    (when on-change-text
+                                      (on-change-text ""))
+                                    (when on-clear
+                                      (on-clear)))
+                                  [focused?])
+        on-scan                  (when on-scan (rn/use-callback #(on-scan set-value)))
+        on-focus                 (rn/use-callback
+                                  (fn []
+                                    (when (= (count @value) 0)
+                                      (set-status :active))
+                                    (set-focused true)
+                                    (when on-focus (on-focus))))
+        on-blur                  (rn/use-callback
+                                  (fn []
+                                    (when (= status :active)
+                                      (set-status :default))
+                                    (set-focused false)
+                                    (when on-blur (on-blur)))
+                                  [status])
+        placeholder-text-color   (rn/use-memo #(get-placeholder-text-color status theme blur?)
+                                              [status theme blur?])]
+    (rn/use-mount #(on-change (or default-value "")))
```

Comment:
Seems like this part is different from how it behaved before. Wondering if there are cases where the input value would have to update due to a change in `default-value`/`scanned-value` and why it was done this way in the first place?

Reply:
default-value can't change, its name says that, scanned-value replaced by proper implementation with  on-scan parameter

Reply:
i see, makes sense now :+1: . I'd say that for the future would be best to separate migration changes from the refactors (e.g. removing use-effects or changing the component behavior). Otherwise it gets difficult to review and I'd guess difficult for QA to check as well. What do you think about separate issues for the components that need more involved refactors?